core/src/main/java/org/infinispan/marshall/core/internal/ExternalJBossMarshaller.java

```diff
@@ -0,0 +1,226 @@
+      marshaller.start();
+   }
+
+   @Override
+   public void stop() {
+      marshaller.stop();
+   }
+
+   @Override
+   public boolean isMarshallable(Object o) throws Exception {
+      return marshaller.isMarshallable(o);
+   }
+
+   @Override
+   public ObjectOutput startObjectOutput(OutputStream os, boolean isReentrant, int estimatedSize) throws IOException {
+      throw new RuntimeException("NYI");
+   }
+
+   @Override
+   public void finishObjectOutput(ObjectOutput oo) {
+      throw new RuntimeException("NYI");
+   }
+
+   @Override
+   public ObjectInput startObjectInput(InputStream is, boolean isReentrant) throws IOException {
+      throw new RuntimeException("NYI");
+   }
+
+   @Override
+   public void finishObjectInput(ObjectInput oi) {
+      throw new RuntimeException("NYI");
+   }
+
+   @Override
+   public Object objectFromInputStream(InputStream is) throws IOException, ClassNotFoundException {
+      throw new RuntimeException("NYI");
+   }
+
+   @Override
+   public byte[] objectToByteBuffer(Object obj, int estimatedSize) throws IOException, InterruptedException {
+      throw new RuntimeException("NYI");
+   }
+
+   @Override
+   public byte[] objectToByteBuffer(Object obj) throws IOException, InterruptedException {
+      throw new RuntimeException("NYI");
+   }
+
+   @Override
+   public Object objectFromByteBuffer(byte[] buf) throws IOException, ClassNotFoundException {
+      throw new RuntimeException("NYI");
+   }
+
+   @Override
+   public Object objectFromByteBuffer(byte[] buf, int offset, int length) throws IOException, ClassNotFoundException {
+      throw new RuntimeException("NYI");
+   }
+
+   @Override
+   public ByteBuffer objectToBuffer(Object o) throws IOException, InterruptedException {
+      throw new RuntimeException("NYI");
+   }
+
+   @Override
+   public BufferSizePredictor getBufferSizePredictor(Object o) {
+      throw new RuntimeException("NYI");
+   }
+
+   static final class JBossByteOutput extends OutputStream implements ByteOutput {
+
+      final ObjectOutput out;
+
+      JBossByteOutput(ObjectOutput out) {
+         this.out = out;
+      }
+
+      @Override
+      public void write(int b) throws IOException {
+         out.write(b);
+      }
+
+   }
+
+   static final class JBossByteInput extends InputStream implements ByteInput {
+
+      final ObjectInput in;
+
+      JBossByteInput(ObjectInput in) {
+         this.in = in;
+      }
+
+      @Override
+      public int read() throws IOException {
+         try {
+            return in.readUnsignedByte();
+         } catch (ArrayIndexOutOfBoundsException e) {
+            // When JBoss Marshalling starts reading a stream, it buffers
+            // the contents, so it'll try as read as much as it can, so it
+            // can end up reading too far, in which case launder the exception
+            // into an IOException so that reading can stop without causing
+            // issues.
+            throw new EOFException();
```

Comment:
The `ByteInput` javadoc says " If no byte is available because the end of the stream has been reached, the value -1 is returned."